mono/utils/mono-time.c

```diff
@@ -126,16 +139,66 @@ get_boot_time (void)
 }
 
 /* Returns the number of milliseconds from boot time: this should be monotonic */
+/* Adapted from CoreCLR: https://github.com/dotnet/coreclr/blob/66d2738ea96fcce753dec1370e79a0c78f7b6adb/src/pal/src/misc/time.cpp */
 gint64
 mono_msec_boottime (void)
 {
-	static gint64 boot_time = 0;
-	gint64 now;
-	if (!boot_time)
-		boot_time = get_boot_time ();
-	now = mono_100ns_datetime ();
-	/*printf ("now: %llu (boot: %llu) ticks: %llu\n", (gint64)now, (gint64)boot_time, (gint64)(now - boot_time));*/
-	return (now - boot_time)/10000;
+    uint64_t retval = 0;
+
+#if HAVE_CLOCK_MONOTONIC_COARSE || HAVE_CLOCK_MONOTONIC
```

Comment:
Where are the configure.ac changes to detect all this new stuff?

Reply:
WIP folks :) we already have `HAVE_CLOCK_MONOTONIC` so the for the fix we want this should be fine.

Reply:
@kumpera lol nope, I thought we had it as I was grepping in Ludo's System.Native integration branch where he brought them over. I'll add them :)